src/statement.ts

```diff
@@ -109,6 +109,17 @@ export class Statement<S = unknown> {
     return { type, sql, tables };
   }
 
+  async #execAndReturn<T = unknown>(
+    params: ExtractedStatement,
+    perfStart: number
+  ): Promise<Result<T>> {
+    const receipt = await checkWait(
+      this.config,
+      await exec(this.config, params)
+    );
+    return wrapResult<T>(receipt, performance.now() - perfStart);
```

Comment:
This isn't related to this PR, but I noticed we repeat this code more than once.